### ISSUE TITLE
DevelopmentCodeFragment doesn't work as expected

### ISSUE THREAD

**Issue by SanderSander** [phpmd/phpmd#949]
- PHPMD version: 2.12.0
- PHP Version: 8.1
- Installation type: (composer)

Function calls like `var_dump` are not detected in by the the Rule/Class `DevelopmentCodeFragment`.
It sees the calls as in my case `string(21) "composerlink\var_dump"` where `composerlink` is the root namespace.

This has to do with the namespace, but in the namespace `composerlink` a method `var_dump` doesn't exist at all.

PHP Just uses the internal `var_dump` method, and you should expect phpmd to fail on this, excepts when you define a method `var_dump` in that namespace. (When `ignore-namespace` is true it should always fail)

### Current Behavior
No violations detected

### Expected Behavior
```
 86 | VIOLATION | The method LinkManager::linkPackage() calls the typical debug function var_dump() which is mostly only used during development.
```

### Steps To Reproduce:
Add `var_dump` somewhere in your code, and run `./vendor/bin/phpmd ./src ansi design` 

## Additional info

With the following configuration it works like excepted:

```
    <rule ref="rulesets/design.xml">
        <exclude name="DevelopmentCodeFragment" />
    </rule>
    <rule ref="rulesets/design.xml/DevelopmentCodeFragment">
        <properties>
            <property name="ignore-namespaces" value="true" />
        </properties>
    </rule>
 ```

## Checks before submitting
* [x] Be sure that there isn't already an issue about this. See: [Issues list](https://github.com/phpmd/phpmd/issues)
* [x] Be sure that there isn't already a pull request about this. See: [Pull requests](https://github.com/phpmd/phpmd/pulls)
* [x ] I have added every step to reproduce the bug.
* [x] If possible I added relevant code examples.
* [x] This issue is about 1 bug and nothing more.
* [x] The issue has a descriptive title. For example:  "JSON rendering failed on Windows for filenames with space".

**Comment by SanderSander** [comment]
I tried to fix it but I''m not aware of a way to detect if the method truly exists
